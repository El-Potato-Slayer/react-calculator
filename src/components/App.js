import { Component, Fragment } from 'react';
import '../App.css';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

class App extends Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = (buttonName) => {
    const { total, next, operation } = this.state;
    const result = calculate({ total, next, operation }, buttonName);
    this.setState({ total: result.total, next: result.next, operation: result.operation });
  }

  render() {
    const data = this.state;

    return (
      <>
        <Display data={data} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
