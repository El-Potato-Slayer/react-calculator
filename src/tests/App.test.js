import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../components/App';

it('Renders App component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
