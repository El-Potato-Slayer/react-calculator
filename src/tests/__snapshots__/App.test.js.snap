// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <nav
    className="navbar"
    data-testid="navbar"
  >
    <p>
      Mathematics Meow Meow
    </p>
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote Of The Day
      </a>
    </div>
  </nav>,
  <div
    className="home"
  >
    <h3>
      Welcome to our page!
    </h3>
    <p>
      We're no strangers to love
       
      <br />
      You know the rules and so do I
       
      <br />
      A full commitment's what I'm thinking of
       
      <br />
      You wouldn't get this from any other guy
       
      <br />
      <br />
      <br />
      I just wanna tell you how I'm feeling
       
      <br />
      Gotta make you understand
       
      <br />
      <br />
      <br />
      Never gonna give you up
       
      <br />
      Never gonna let you down
       
      <br />
      Never gonna run around and desert you
       
      <br />
      Never gonna make you cry
       
      <br />
      Never gonna say goodbye
       
      <br />
      Never gonna tell a lie and hurt you
       
      <br />
      <br />
      <br />
      We've known each other for so long
       
      <br />
      Your heart's been aching but you're too shy to say it
       
      <br />
      Inside we both know what's been going on
       
      <br />
      We know the game and we're gonna play it
       
      <br />
      And if you ask me how I'm feeling
       
      <br />
      Don't tell me you're too blind to see
       
      <br />
      <br />
      <br />
      Never gonna give you up
       
      <br />
      Never gonna let you down
       
      <br />
      Never gonna run around and desert you
       
      <br />
      Never gonna make you cry
       
      <br />
      Never gonna say goodbye
       
      <br />
      Never gonna tell a lie and hurt you
       
      <br />
      <br />
      <br />
      Never gonna give you up
       
      <br />
      Never gonna let you down
       
      <br />
      Never gonna run around and desert you
       
      <br />
      Never gonna make you cry
       
      <br />
      Never gonna say goodbye
       
      <br />
      Never gonna tell a lie and hurt you
       
      <br />
      <br />
      <br />
      Never gonna give, never gonna give
       
      <br />
      (Give you up)
       
      <br />
      <br />
      <br />
      We've known each other for so long
       
      <br />
      Your heart's been aching but you're too shy to say it
       
      <br />
      Inside we both know what's been going on
       
      <br />
      We know the game and we're gonna play it
       
      <br />
      <br />
      <br />
      I just wanna tell you how I'm feeling
       
      <br />
      Gotta make you understand
       
      <br />
      <br />
      <br />
      Never gonna give you up
       
      <br />
      Never gonna let you down
       
      <br />
      Never gonna run around and desert you
       
      <br />
      Never gonna make you cry
       
      <br />
      Never gonna say goodbye
       
      <br />
      Never gonna tell a lie and hurt you
       
      <br />
      <br />
      <br />
      Never gonna give you up
       
      <br />
      Never gonna let you down
       
      <br />
      Never gonna run around and desert you
       
      <br />
      Never gonna make you cry
       
      <br />
      Never gonna say goodbye
       
      <br />
      Never gonna tell a lie and hurt you
       
      <br />
      <br />
      <br />
      Never gonna give you up
       
      <br />
      Never gonna let you down
       
      <br />
      Never gonna run around and desert you
       
      <br />
      Never gonna make you cry
       
      <br />
      Never gonna say goodbye
       
      <br />
    </p>
  </div>,
]
`;
